swim/node_bootstrap.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2015 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package swim
+
+import (
+	"encoding/json"
+	"io/ioutil"
+)
+
+// A HostProvider is a interface that provides a list of hosts for a node to
+// bootstrap from.
+type HostProvider interface {
+	Hosts() ([]string, error)
+}
+
+// JSONFileHostProvider is a HostProvider that reads a list of hosts from a
+// JSON file.
+type JSONFileHostProvider struct {
+	filePath string
+}
+
+// NewBootstrapFileHostProvider returns a new JSONFileHostProvider.
+func NewJSONFileHostProvider(filePath string) (*JSONFileHostProvider, error) {
```

Comment:
Should this constructor be removed? It's not strictly necessary. Also there is no possible error condition. In retrospect, this constructor seems pointless.